internal/datasource/config/config_writer.go

```diff
@@ -60,15 +61,32 @@ func (cw *ConfigWriter) SetConfigClient(configClient client.ConfigClientInterfac
 	cw.configClient = configClient
 }
 
+func (cw *ConfigWriter) Rollback(ctx context.Context, skippedFiles CacheContent, filesURL,
+	tenantID, instanceID string,
+) error {
+	slog.Debug("Rolling back NGINX config changes due to error")
+	for key, value := range cw.fileCache.CacheContent() {
+		if _, ok := skippedFiles[key]; ok {
+			continue
+		}
+
+		err := cw.updateFile(ctx, value, filesURL, tenantID, instanceID)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
 func (cw *ConfigWriter) Write(ctx context.Context, filesURL,
 	tenantID, instanceID string,
-) (skippedFiles map[string]struct{}, err error) {
+) (skippedFiles CacheContent, err error) {
+	slog.Info("Writing file")
```

Comment:
Should this be Debug?

Reply:
Also maybe add filesURL to the message so that we know what file is being written